Week3/homework/step2-6.js

```diff
@@ -4,18 +4,28 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const result = arr.reduce((acc, x) => acc.concat(x));
+  return result;
```

Comment:
I haven't seen this approach in previous classes, but yes, it works fine. Note that, since you are not using the variable `result` any further in your code you might as well return the value directly.

```js
return arr.reduce((acc, x) => acc.concat(x));
```